### ISSUE TITLE
[bug] windows: shuffle sequence is not random & repeats upon every app start-up

### ISSUE THREAD

**Issue by KyleKun** [harmonoid/harmonoid#308]
**Describe the bug**
When Random is enabled, songs are repeating the same sequence everytime.

**To Reproduce**
1. Go to Tracks
2. Click on the first song
3. Click on the Random button
4. Go to the next songs
5. Close the app and repeat the steps, the sequence of songs is the same

**Expected behavior**
Songs sequence should be different instead of repeating.

**Desktop :**
 - OS: Windows 11
 - Version 0.2.6

**Comment by alexmercerind** [comment]
Hi!
I'm unfortunately unable to confirm the reported issue. 

https://user-images.githubusercontent.com/28951144/182689563-e2e68c8b-1c5d-4e28-a357-023cfa401b54.mp4

Let me know if you intend shuffle to be working in some other way or if I'm missing something about the bug.
Thanks!

**Comment by KyleKun** [comment]
Hi @alexmercerind , thanks for the quick reply!
Check this video on how to reproduce.

I go to Tracks and select the first song on the top, click the shuffle button and advance to the next songs.
Then after closing the program and doing the same again, the sequence of songs is still the same.

https://user-images.githubusercontent.com/36611739/182695298-536f706e-b8f1-4682-bd06-9d0f0075c64c.mp4

**Comment by alexmercerind** [comment]
Hi @KyleKun!
Thanks for the descriptive video, unfortunately the issue still doesn't reproduce on my machine.

I'm pinning this issue to draw attention of other people who may be experiencing the same issue & are willing to provide more information.

<hr>

Apart from this issue, I notice in your video, that `Year`  tag for some of your songs is showing date instead of year.
That will be addressed in the next release version.

Thanks!!!

**Comment by KyleKun** [comment]
Hi @alexmercerind, I was digging into the code and it seems to me that the issue is in `libmpv.dart`, but I couldn't find the repo. 

I guess [this was a fork](https://github.com/FireMasterK/libmpv.dart/blob/master/lib/src/player.dart#L390), so if the code is the same as today, the `set shuffle` method seems to get files' info to set the songs order, so I presume something in my files properties trigger this behavior. Maybe this part inside `set shuffle` can give us a hint:

``` dart
final property = data
                      .ref.u.list.ref.values[i].u.list.ref.keys[j]
                      .cast<Utf8>()
                      .toDartString();
                  if (property == 'filename') {
                    final value = data
                        .ref.u.list.ref.values[i].u.list.ref.values[j].u.string
                        .cast<Utf8>()
                        .toDartString();
                    playlist.add(medias[value]!);

```

**Comment by alexmercerind** [comment]
> Hi @alexmercerind, I was digging into the code and it seems to me that the issue is in `libmpv.dart`, but I couldn't find the repo.
> 

Hey @KyleKun! Sorry for bothering you again.
Could you please start the application from command prompt like this:

```
C:\Program Files (x86)\Harmonoid>harmonoid.exe > log.txt
```

This will log out the terminal output from the executable into the `log.txt`. It may contain some necessary information for be to actually find out the cause. A lot of [libmpv.dart](https://github.com/harmonoid/libmpv.dart) has been changed since & addressed. Let's see what's happening in there.

Thanks!

**Comment by KyleKun** [comment]
Hey @alexmercerind, sure thing, this is `log.txt` output:
```
flutter: _tracks: 3953
flutter: _albums: 722
flutter: _artists: 534
flutter: scaletempo:scale=1.00000000
flutter: scaletempo:scale=1.00000000
```

And the link to `libmpv.dart` is not working btw (404 not found) :(

**Comment by alexmercerind** [comment]
> Hey @alexmercerind, sure thing, this is `log.txt` output:
> 
> ```
> flutter: _tracks: 3953
> flutter: _albums: 722
> flutter: _artists: 534
> flutter: scaletempo:scale=1.00000000
> flutter: scaletempo:scale=1.00000000
> ```
> 
> And the link to `libmpv.dart` is not working btw (404 not found) :(

Hey thanks!

Does the log contain some more stuff after pressing/un-pressing the shuffle button a couple of times (make sure you have a playlist loaded).

Thanks again!

**Comment by KyleKun** [comment]
> > Hey @alexmercerind, sure thing, this is `log.txt` output:
> > ```
> > flutter: _tracks: 3953
> > flutter: _albums: 722
> > flutter: _artists: 534
> > flutter: scaletempo:scale=1.00000000
> > flutter: scaletempo:scale=1.00000000
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And the link to `libmpv.dart` is not working btw (404 not found) :(
> 
> Hey thanks!
> 
> Does the log contain some more stuff after pressing/un-pressing the shuffle button a couple of times (make sure you have a playlist loaded).
> 
> Thanks again!

Nope, that's all the content no matter what I do in the app.

**Comment by alexmercerind** [comment]
> > > Hey @alexmercerind, sure thing, this is `log.txt` output:
> > > ```
> > > flutter: _tracks: 3953
> > > flutter: _albums: 722
> > > flutter: _artists: 534
> > > flutter: scaletempo:scale=1.00000000
> > > flutter: scaletempo:scale=1.00000000
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > And the link to `libmpv.dart` is not working btw (404 not found) :(
> > 
> > 
> > Hey thanks!
> > Does the log contain some more stuff after pressing/un-pressing the shuffle button a couple of times (make sure you have a playlist loaded).
> > Thanks again!
> 
> Nope, that's all the content no matter what I do in the app.

Thanks a lot again!

**Comment by alexmercerind** [comment]
Shall be addressed in latest release: [v0.2.9](https://github.com/harmonoid/harmonoid/releases/tag/v0.2.9).

I've majorly re-written playlist management to make things compatible with mpv versions v0.32.x or lower.
There hasn't been any such report from someone else, closing.

**Comment by alexmercerind** [comment]
Sorry for the inconvenience, the issue indeed exists for Windows.

**Comment by alexmercerind** [comment]
Oh I forgot to close this, it should be fixed in latest release: [v0.3.6](https://github.com/harmonoid/harmonoid/releases/tag/v0.3.6).

This was an issue within libmpv. C's `srand` works per-thread on Windows apparently. Now libmpv uses it's own random number implementation, thus shuffle should work similarly & correctly on all platforms. Further information: https://github.com/mpv-player/mpv/pull/10528.
